### PR TITLE
GridMapping / Spatial Resampling

### LINKED ISSUE #396: Unittest test_fetch_wmts_kvp_tile does not pass anymore
**Describe the bug**
Unittest test_fetch_wmts_kvp_tile does not pass anymore. The error message is `Illegal format 'image/png'`

See:
https://ci.appveyor.com/project/bcdev/xcube/builds/37132711/job/a5oatfkbfhs4oimk#L2400

### LINKED ISSUE #400: Missing files for executing example notebooks
For the following notebooks files are missing which are needed as input: 

- [ ] xcube/examples/notebooks/rectify_sen3_olci.ipynb `[Errno 2] No such file or directory: 'C:\\Users\\Norman\\Downloads\\S3B_OL_1_EFR____20190728T103451_20190728T103751_20190729T141105_0179_028_108_1800_LN1_O_NT_002.SEN3'`
- [ ] xcube/examples/notebooks/rectify_sen3_olci_ex.ipynb `[Errno 2] No such file or directory: 'C:\\Users\\Norman\\Downloads\\S3B_OL_1_EFR____20190728T103451_20190728T103751_20190729T141105_0179_028_108_1800_LN1_O_NT_002.SEN3'`
- [x] xcube/examples/notebooks/rectify.ipynb `[Errno 2] No such file or directory: b'/Users/appveyor/projects/notebooks/xcube-gen-bc/test/inputdata/O_L2_0001_SNS_2017104102450_v1.0.nc'`

Please add the necessary files and adjust the paths in the notebooks.

### PR DESCRIPTION
**Pull request by forman**
This is part one of two of a large breaking change in xcube that will allow for generic spatio-temporal resampling (#391). This part one concentrates on the implementation of spatial resampling. Part 2 will focus on combined  spatio-temporal resampling and integration into the xcube gen2 tool.

### Issues

- Addresses #391
- Addresses #112
- Addresses #484
- Fixes #400
- Fixes #396

### Major changes:

* Numerous breaking changes have been applied to this version in order to address generic resampling (#391) and support other CRS than WGS-84 (#112): 
  * The following components have been removed entirely 
    - module `xcube.core.imgeom` with class `ImageGeom` 
    - module `xcube.core.geocoding` with class `GeoCoding`
    - module `xcube.core.reproject` and all its functions
  * The following components have been added 
    - module `xcube.core.gridmapping` with new class `GridMapping`
      is a CF compliant replacement for classes `ImageGeom` and `GeoCoding`
  * The following components have changed in an incompatible way:
    - Function`xcube.core.rectify.rectify_dataset()` now uses 
      `source_gm: GridMapping` and `target_gm: GridMapping` instead of 
      `geo_coding: GeoCoding` and `output_geom: ImageGeom`. 
    - Function`xcube.core.gen.iproc.InputProcessor.process()` now uses 
      `source_gm: GridMapping` and `target_gm: GridMapping` instead of 
      `geo_coding: GeoCoding` and `output_geom: ImageGeom`. 
  * xcube no longer depends on GDAL (at least not directly).
 

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* ~[ ] New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] AppVeyor passes
* [x] Test coverage remains or increases (target 100%)

Remember to close associated issues after merge!

### CODE REVIEW

**Review by AliceBalfanz — COMMENTED** [review]
please view failing build: 
=========================================================================================================== short test summary info ============================================================================================================
FAILED test/core/gen2/test_generator.py::LocalCubeGeneratorTest::test_generate_cube_from_dict - ValueError: too many values to unpack (expected 2)
FAILED test/core/gen2/test_generator.py::LocalCubeGeneratorTest::test_generate_cube_from_json - ValueError: too many values to unpack (expected 2)
FAILED test/core/gen2/test_generator.py::LocalCubeGeneratorTest::test_generate_cube_from_yaml - ValueError: too many values to unpack (expected 2)
===================================================================================== 3 failed, 967 passed, 10 skipped, 590 warnings in 328.60s (0:05:28) ======================================================================================
Command exited with code 1
Build failed

Otherwise approving

**Review by pont-us — APPROVED** [review]

**Review by TonioF — CHANGES_REQUESTED** [review]
As we now use positional arguments, the minimum python version should be increased to 3.8

File: xcube/core/gridmapping/regular.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+from .helpers import _default_xy_var_names
+from .helpers import _normalize_int_pair
+from .helpers import _normalize_number_pair
+from .helpers import _to_int_or_float
+
+
+class RegularGridMapping(GridMapping):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
```
[the remark this answers is not in the document]
Comment (forman) [reply xcube/core/gridmapping/regular.py:41]: This is an internal class, and is not intended to be invoked from outside its package.

**Comment by forman** [comment]
> As we now use positional arguments, the minimum python version should be increased to 3.8

Agreed. I'll change that.

**Pull request merged**